tools/src/main/java/org/apache/kafka/trogdor/workload/ProduceBenchWorker.java

```diff
@@ -84,15 +90,32 @@ public void start(Platform platform, WorkerStatusTracker status,
             throw new IllegalStateException("ProducerBenchWorker is already running.");
         }
         log.info("{}: Activating ProduceBenchWorker with {}", id, spec);
-        // Create an executor with 2 threads.  We need the second thread so
-        // that the StatusUpdater can run in parallel with SendRecords.
-        this.executor = Executors.newScheduledThreadPool(2,
+        // Create an executor with 3 threads.  We need the second thread so
+        // that the StatusUpdater can run in parallel with SendRecords, and
+        // the third thread is for refreshing the partition info, if desired.
+        this.executor = Executors.newScheduledThreadPool(3,
             ThreadUtils.createThreadFactory("ProduceBenchWorkerThread%d", false));
         this.status = status;
         this.doneFuture = doneFuture;
         executor.submit(new Prepare());
     }
 
+    private HashSet<TopicPartition> getActivePartitions() throws InterruptedException, ExecutionException {
```

Comment:
nit: Any reason to not return the more general <Set>?

Reply:
I wanted the PartitionRefresher class to be as simple as possible, so I decided to push all logic here.